Removes dependency on QtX11Extras for Qt6.2-beta

spiderkeys opened the pull request:
Qt6 has deprecated QtX11Extras and intends to move that functionality into QtGUI's platform code:
https://github.com/qt/qtx11extras

As of the latest 6.2-Beta release available as prebuilt binaries, the QtX11Extras functionality has been added as a private API, but there is still not a public API available:
https://codereview.qt-project.org/c/qt/qtbase/+/348530

This PR seeks to introduce a workaround for Qt6 that removes the external QtX11Extra dependency by implementing the two calls that ADS depends on directly in ads_globals.cpp as a workaround until the new public API is available.

@githubuser0xFFFF I know there probably need to be a few Qt6 specific version checks, but I wasn't sure how you would prefer this to be done, or if you approve of this method of workaround at all. Let me know your thoughts and I will be happy to make changes or close this PR altogether.

githubuser0xFFFF commented:
Does your solution also work for Qt5? If it works for Qt5 then I would accept the pull request to get rid of QtX11Extras.

spiderkeys commented:
Hi @githubuser0xFFFF I tested with both 5.15.4 and 6.2.0-beta and both seem to work fine, but I would encourage you to test it out as well on any versions you might have concerns about. I'm not sure exactly how far back the use of `#include <qpa/qplatformnativeinterface.h>` works...

I've also cleaned up the code a bit and also updated the CI-related bits that installed/mentioned x11extras.

Let me know if there is anything else you would like changed.

spiderkeys commented on src/CMakeLists.txt:
```diff
@@ -64,10 +64,6 @@ endif()
                                                Qt${QT_VERSION_MAJOR}::Widgets)
-if(UNIX AND NOT APPLE)
-    target_link_libraries(qtadvanceddocking PUBLIC Qt${QT_VERSION_MAJOR}::X11Extras)
-	target_link_libraries(qtadvanceddocking PRIVATE xcb)
```
This doesn't seem to be explicitly needed, as it worked in my tests for 5.15.4, but not sure if that is the case for pre-5.15 releases.

githubuser0xFFFF commented:
Hi,

I just tested your pull request on Ubuntu 18.04 with Qt 5.9.5 and it works - great. The only thing I had to change was the src.pro file, because I use qmake build. I added **gui-private** to the Qt variable:

``` qmake
unix:!macx {
HEADERS += linux/FloatingWidgetTitleBar.h
SOURCES += linux/FloatingWidgetTitleBar.cpp
LIBS += -lxcb
QT += gui-private
}
```

After this change I could build and run the demo application. I will also test this with Ubuntu 16.04, to check if this also works.

Thank you.

spiderkeys commented:
src.pro has been updated.

Also noticed that there was an open issue relating to this, so linking here just to connect the PR : https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/issues/324

ruilvo commented:
I'd appreciate for this to get merged :)

githubuser0xFFFF commented:
@ruilvo Could you please test, if this change works on Ubuntu 16.04. As soon as I know that it works on Ubuntu 16.04 I will merge this.
